addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -219,26 +219,82 @@ def _prepare_down_payment_section_values(self, order):
         del context
         return so_values
 
-    def _prepare_so_line_values(self, order):
+    def _prepare_so_lines_values(self, order):
+        self.ensure_one()
+        down_payment_amount = self._get_down_payment_amount(order)
+        group_by_taxes = defaultdict(lambda: {
```

Comment:
you must never do such kind of computation yourself. Instead, use the helpers on `account.tax`. For example:
https://github.com/odoo/odoo/blob/16.0/addons/sale/models/sale_order_line.py#L598

This allows you to get the tax amounts (grouped or not depending the keys you want) for any business documents

Reply:
Did that + simplified

Reply:
I think you can simplify more using `_aggregate_taxes` by passing a custom `grouping_key_generator`